core/trino-main/src/main/java/io/trino/sql/planner/optimizations/BeginTableWrite.java

```diff
@@ -263,20 +298,17 @@ private PlanNode rewriteModifyTableScan(PlanNode node, TableHandle handle)
 
     public static class Context
```

Comment:
Since the `Context` now just holds an `Optional<WriterTarget>`, the visitor could just use `Optional<WriterTarget>` as the context type and get rid of this class.

Reply:
`Optional` is not mutable. We could use `AtomicReference<WriterTarget>`, but would still need a place to put the sanity checks that are currently in the `Context` class, so I don't think that's an improvement.

Reply:
It doesn't need to be mutable. You just need to pass the proper instance when visiting the child.

Reply:
Where it does 

```java
child = child.accept(this, context);
```

you'd do this instead:

```java
child = child.accept(this, Optional.of(newTarget);
```

In the `optimize()` method:

```java
return SimplePlanRewriter.rewriteWith(new Rewriter(session), plan, Optional.empty());
```

Reply:
Makes sense. I'll update it